internal/pkg/config/loader.go

```diff
@@ -0,0 +1,113 @@
+package config
+
+import (
+	"flag"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"os"
+	"strconv"
+	"time"
+)
+
+func Load(path string) (*ServiceConfig, error) {
+	// Load default config from disk
+	viper.SetConfigName("service")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath("/configs/")
+	if path != "" {
+		viper.AddConfigPath(path)
+	}
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			log.Error("cannot find config file, aborting")
+		} else {
+			log.Error("error reading config file, aborting")
+		}
+		return nil, err
+	}
+
+	// Environment variable overrides
+	viper.AutomaticEnv()
+	if err := viper.BindEnv(RedisAddress, "OPEN_SAVES_CACHE", "REDIS_ADDRESS"); err != nil {
+		log.Warning("cannot bind env var %s with %s", "OPEN_SAVES_CACHE", RedisAddress)
+	}
+
+	// Reads command line arguments, for backward compatibility
```

Comment:
Do you think we can use the viper's flag binding here
https://github.com/spf13/viper#working-with-flags

Reply:
I have looked into it but the issue is that the flags we define differ from the actual config key name. So there does not seem to be a way to say "bind this flag to this config value" like I have done for thr environment variable OPEN_SAVES_CACHE.

Unless we want to switch to using the pflag module of viper, I'd leave it as is. But even then there would not be any value in doing so.

Reply:
I think we can do something like

``` go
pflag.Int("port", 6000, "The port number to run Open Saves on")
// ...
viper.BindPFlag(OpenSavesPort, pflag.Lookup("port"));
```

Reply:
I'll give this a try.